core/model-api/src/main/java/org/eclipse/rdf4j/model/Namespace.java

```diff
@@ -22,9 +33,23 @@
 	public String getName();
 
 	/**
-	 * Gets the prefix of the current namespace. The default namespace is represented by an empty prefix string.
+	 * Compares this namespace to another object.
 	 *
-	 * @return prefix of namespace, or an empty string in case of the default namespace.
+	 * @param o The object to compare this namespace to
+	 *
+	 * @return {@code true} if the other object is an instance of {@code Namespace} and their {@linkplain #getPrefix()
+	 *         prefixes} and {@linkplain #getName() names} are equal, {@code false} otherwise.
 	 */
-	public String getPrefix();
+	@Override
+	boolean equals(Object o);
+
+	/**
+	 * Computes the hash code of this namespace.
+	 *
+	 * @return a hash code for this namespace computed as {@link java.util.Objects#hash Objects.hash}(
+	 *         {@link #getPrefix()}, {@link #getName()})
+	 */
+	@Override
+	int hashCode();
```

Comment:
This, also, is not backwards compatible. I don't think we need to declare these at the interface level right now to get this to work.

Reply:
I still blame lack of coffee. This is not a problem in the sense that any subclass that does not explicitly implement these methods will just inherit them from Object. Which is not correct according the documented exepcted behavior we give here, but at least it won't cause compilation errors.

Reply:
I had a closer look at this and I think that in fact you are correct to specify this, and we should treat this not as a backward-incompatible change, but as a bug fix. Namespace extends Comparable, which _requires_ that we have properly defined contract for equals / hashCode. 

I'll add the necessary exceptions to the api comparison check.